### ISSUE TITLE
MIT vs LGPL license conflict?

### ISSUE THREAD

**Issue by whot** [libratbag/ratbagd#23]
We have the rbtree implementation which is LGPL but otherwise ratbagd is MIT. Problem is - since we use the rbtree directly rather than as runtime-linked library, I'm pretty sure this would mean ratbagd becomes LGPL too. In which case, having it as MIT is a bit pointless, since it's always linked in.

Quick skim of the sources suggests we can just replace the rbtree with a linked list, given the normal number of devices we're dealing with the rbtree feels like overkill anyway.

**Comment by bentiss** [comment]
I am fine replacing it with a linked-list or changing the license to LGPL.

**Comment by dvdhrm** [comment]
I don't like using lists where they're not appropriate, which is why I added the rb-tree implementation. The code is from http://github.com/c-util/c-rbtree and is ASL-2.0 these days. So feel free to update the code, or just change the license.

Btw., statically linking LGPL code is mostly fine, but you now need to provide users with the possibility to re-link with a newer version of the LGPL'ed code. For binary distributions this means either always distributing the source, or if that's not appropriate then at least distribute the object-files unlinked.

Long story short: Feel free to drop the license header from shared-rbtree.[ch], or replace it with a linked list, if you prefer. I can approve both.

**Comment by whot** [comment]
did you write the rbtree from scratch? if so, wouldn't the easiest be to just make it MIT or ratbagd? Or do you care sufficiently about the license in those files? I don't know the ASL vs MIT interactions and this way we don't have remember to do dual-licensing tags in the variuos distributions.

Linked list - in 99% of use-cases we'll have a single element, so that linked list is likely to be just a fancy pointer anyway :)

**Comment by dvdhrm** [comment]
Yes, I wrote the rbtree from scratch, and I have no strong feelings regarding the licenses. Pick something and I will approve it. Or drop it, if you don't have any strong feelings on the matter ;)
